shimmingtoolbox/coils/siemens_basis.py

```diff
@@ -0,0 +1,154 @@
+        (Requires checking the Adjustments/Shim card to see what the corresponding terms and values actually are). So,
+        for now, ``basis`` will always be returned with 8 terms along the 4th dim.
+    """
+
+    # Local functions
+    def reorder_to_siemens(spher_harm):
+        """
+        Reorder 1st - 2nd order basis terms along 4th dim. From
+        1. Y, Z, X, XY, ZY, Z2, ZX, X2 - Y2 (output by shimmingtoolbox.coils.spherical_harmonics.spherical_harmonics), to
+        2. X, Y, Z, Z2, ZX, ZY, X2 - Y2, XY (in line with Siemens shims)
+
+        Args:
+            spher_harm (numpy.ndarray): 4d basis set of spherical harmonics with order/degree ordered along 4th
+                                        dimension. ``spher_harm.shape[3]` must equal 8.
+
+        Returns:
+            numpy.ndarray: 4d basis set of spherical harmonics ordered following siemens convention
+        """
+
+        assert spher_harm.shape[3] == 8
+        reordered = np.zeros(spher_harm.shape)
+        reordered[:, :, :, 0] = spher_harm[:, :, :, 2]
+        reordered[:, :, :, 1] = spher_harm[:, :, :, 0]
+        reordered[:, :, :, 2] = spher_harm[:, :, :, 1]
+        reordered[:, :, :, 3] = spher_harm[:, :, :, 5]
+        reordered[:, :, :, 4] = spher_harm[:, :, :, 6]
+        reordered[:, :, :, 5] = spher_harm[:, :, :, 4]
+        reordered[:, :, :, 6] = spher_harm[:, :, :, 7]
+        reordered[:, :, :, 7] = spher_harm[:, :, :, 3]
+
+        return reordered
+
+    def get_scaling_factors():
+        """
+        Get scaling factors for the 8 terms to apply to the (Siemens-reordered) 1st + 2nd order spherical harmonic
+        fields for rescaling field terms as "shim reference maps" in units of Hz/unit-shim:
+
+        Gx, Gy, and Gz should yield 1 micro-T of field shift per metre: equivalently, 0.042576 Hz/mm
+
+        2nd order terms should yield 1 micro-T of field shift per metre-squared: equivalently, 0.000042576 Hz/mm^2
+
+        Gist: given the stated nominal values, we can pick several arbitrary reference positions around the
+        origin/isocenter at which we know what the field *should* be, and use that to calculate the appropriate scaling
+        factor.
+
+        Returns:
+             numpy.ndarray:  1D vector of ``scaling_factors``
+
+        NOTE: The method has been worked out empirically and has only been tested for 2 Siemens Prisma systems.
+        E.g.re: Y, Z, ZX, and XY terms, their polarity had to be flipped relative to the form given by
+        ``shimmingtoolbox.coils.spherical_harmonics``.
+        """
+
+        [x_iso, y_iso, z_iso] = np.meshgrid(np.array(range(-1, 2)), np.array(range(-1, 2)), np.array(range(-1, 2)),
+                                            indexing='xy')
+        orders = np.array(range(1, 3))
+        sh = spherical_harmonics(orders, x_iso, y_iso, z_iso)
+        sh = reorder_to_siemens(sh)
+
+        n_channels = sh.shape[3]
```

Comment:
Is there a need to get the number of channels from the shape since it will always be 8 from reorder_to_siemens function?

Reply:
We will eventually implement this function up to the third order so we will need to check for the size.